docs/Hooks.md

```diff
@@ -141,6 +141,48 @@ fastify.addHook('onSend', (request, reply, payload, next) => {
 
 Note: If you change the payload, you may only change it to a `string`, a `Buffer`, a `stream`, or `null`.
 
+### Respond to a request from an hook
+If need you can respond to a request before you reach the route handler, an example could be an authentication hook. If you are using `onRequest` or a middleware just use `res.end`, if you are using the `preHandler` hook use `reply.send`. Remember to always call `next` if you are using the standard hook api, if you are working with *async* hooks it will be done automatically by Fastify.
+```js
+// standard api
+fastify.addHook('preHandler', (request, reply, next) => {
+  reply.code(401).send(new Error('Unauthorized'))
+  next()
+})
+
+// async api
+fastify.addHook('preHandler', async (request, reply) => {
+  reply.code(401).send(new Error('Unauthorized'))
```

Comment:
A better example would be to `send` something that isn't an Error because you can already do this with `next(new Error('Unauthorized')` for callbacks and `throw new Error('Unauthorized')` for async.